src/agent/plugins.ts

```diff
@@ -41,6 +42,32 @@ export const oid4vciHolder = new OID4VCIHolder({
   onCredentialStored: async (args: OnCredentialStoredArgs): Promise<void> => {
     const {credential, vcHash} = args;
     store.dispatch<any>(dispatchVerifiableCredential(vcHash, credential));
+
+    // FIXME temp solution to have activity for oid4vci-holder, we should add this to the plugin later
+    const contact = store
+      .getState()
+      .contact.contacts.find(contact => contact.identities.some(identity => identity.identifier.correlationId === credential.issuerCorrelationId));
+
+    store.dispatch<any>(
+      storeActivityLogging({
+        level: LogLevel.INFO,
+        system: System.OID4VCI,
+        subSystemType: SubSystem.VC_ISSUER,
+        initiatorType: InitiatorType.SYSTEM,
+        description: 'onCredentialStored event call',
+        actionType: ActionType.CREATE,
+        actionSubType: DefaultActionSubType.VC_ISSUE,
+        diagnosticData: {digitalCredential: credential},
+        // @ts-ignore
+        credentialType: credential.documentFormat, // TODO fix types
+        credentialHash: vcHash,
+        originalCredential: JSON.stringify(credential),
+        // @ts-ignore
+        partyCorrelationType: contact?.identities[0].identifier.type, // TODO fix types
```

Comment:
I think here we are sure that .identities[0] will be present right? (contact.identities.some filter)

Reply:
yes, because every contact has at least 1 or more identities